atom/common/api/atom_api_shell.cc

```diff
@@ -43,17 +44,24 @@ struct Converter<base::win::ShortcutOperation> {
 
 namespace {
 
-void OnOpenExternalFinished(
-    v8::Isolate* isolate,
-    const base::Callback<void(v8::Local<v8::Value>)>& callback,
-    const std::string& error) {
+void OnOpenExternalFinished(const v8::Global<v8::Context>& context,
+                            scoped_refptr<atom::util::Promise> promise,
+                            const std::string& error) {
+  v8::Isolate* isolate = promise->isolate();
```

Comment:
Should this be done in the promise helper to reduce this boilerplate being duplicated for all promises

Reply:
I agree moving the code to promise helper. But I would prefer doing it in a separate PR since it is a refactoring that modifies multiple files.